mypy/test/testdaemon.py

```diff
@@ -0,0 +1,83 @@
+"""End-to-end test cases for the daemon (dmypy).
+
+These are special because they run multiple shell commands.
+"""
+
+import os
+import subprocess
+import sys
+
+from typing import List, Tuple
+
+from mypy.test.config import test_temp_dir, PREFIX
+from mypy.test.data import DataDrivenTestCase, DataSuite
+from mypy.test.helpers import assert_string_arrays_equal
+
+# Files containing test cases descriptions.
+daemon_files = [
+    'daemon.test',
+]
+
+class DaemonSuite(DataSuite):
+    files = daemon_files
+
+    def run_case(self, testcase: DataDrivenTestCase) -> None:
+        test_daemon(testcase)
+
+
+def test_daemon(testcase: DataDrivenTestCase) -> None:
+    if sys.platform == 'win32':
+        return  # These tests don't run on Windows yet.
+    assert testcase.old_cwd is not None, "test was not properly set up"
+    for i, cmd in enumerate(parse_script(testcase.input)):
+        input = cmd[0]
+        expected_lines = cmd[1:]
+        assert input.startswith('$')
+        input = input[1:].strip()
+        if input.startswith('dmypy '):
+            input = 'python3 -m mypy.' + input
```

Comment:
`python3` -> `sys.executable` for Windows.

Reply:
OK. (Though I'd be surprised if that's all that's needed to make this test work on Windows.)

Reply:
Actually, it is! Though it seems that the last test is failing, but I think that might be an issue on my end, so I will experiment a bit more.

Reply:
No, I don't think its that, echo exists on Windows and works very similarly to the echo on Linux. Furthermore, the output of the failing test is empty, which is very wrong. Something else is going on, I'll experiment with it on Monday when I have time to get back to things.